src/formatter.rs

```diff
@@ -408,18 +424,41 @@ mod tests {
 
     #[test]
     fn test_color_formatter_attributes() {
-        // Test that each attribute of the style can be set.
+        // Test that each attribute of the style can be set and that they can be
+        // combined in a single rule or by using multiple rules.
         let config = config_from_string(
             r#"
         colors.fg = { fg = "red" }
+        colors.bg = { bg = "blue" }
+        colors.multiple = { fg = "green", bg = "yellow" }
         "#,
         );
         let mut output: Vec<u8> = vec![];
         let mut formatter = ColorFormatter::for_config(&mut output, &config);
         formatter.add_label("fg").unwrap();
         formatter.write_str(" fg only ").unwrap();
         formatter.remove_label().unwrap();
-        insta::assert_snapshot!(String::from_utf8(output).unwrap(), @"[38;5;1m fg only [39m");
+        formatter.write_str("\n").unwrap();
```

Comment:
It would be nice to have a test somewhere that if `colors."abra cadabra" = "green"`, a single label doesn't make it green and both labels together do, in either order.

Reply:
Thank you! That's nice documentation.

> We actually require the order to match since https://github.com/martinvonz/jj/pull/993.

I see, I missed most of that discussion.